pkg/k8s/kubeconfig/kubeconfig.go

```diff
@@ -60,6 +60,16 @@ func Write(path string, clusterConfig *api.Config, setCurrent bool) error {
 		existingConfig.CurrentContext = clusterConfig.CurrentContext
 	}
 
+	if namespace != "" {
```

Comment:
Do you think its better settings the namespace here? Or when we generate the context [here](https://github.com/arghya88/kconnect/blob/namespace-param/pkg/plugins/discovery/aws/config.go#L45)?

Reply:
Well...that's also an option..I am just trying to think what would be the pros or doing that

Reply:
Same here :)

Reply:
We can probably leave it as is

Reply:
I think we are able to avoid the for loop if we do it there. So to do it there we need to change the function signature and add `params.Namespace`

Reply:
Yes or you could get the config from the ctx.ConfigurationItems(). Feels better as a parameter

Reply:
Its up to you if you want to change it or leave it as is.....whatever you think is best :smile:

Reply:
One disadvantage is that same code needs to copied be for all the providers if we add more providers later.But I guess we can tackle that duplication in a different way in Golang(I am pretty new here). Let's leave it this way for now and refactor it later if we can find a good way to avoid that duplication.

Reply:
Actually let me move it there

Reply:
Sounds like a good plan to me @arghya88